src/wp-includes/rest-api/endpoints/class-wp-rest-block-types-controller.php

```diff
@@ -512,6 +513,134 @@ public function get_item_schema() {
 					'context'     => array( 'embed', 'view', 'edit' ),
 					'readonly'    => true,
 				),
+				'variations'       => array(
```

Comment:
There is some duplication in the schema, many of the items that variations define are shared with the block. Should we account for that to make maintenance simpler and to mark relevance?

@TimothyBJacobs or @spacedmonkey, what would you advise here?

Reply:
In an ideal world we'd have `$ref` support in WordPress Core. Though, since this is a readonly endpoint we could probably document this with `$ref` + `definitions` + `allOf` even though `rest_validate_value_from_schema` wouldn't understand it.

That'd probably be my recommendation at the moment. Alternately, we could pull the shared bits into a PHP variable and just solve the duplication issue that way.

Reply:
I pulled out some reused definitions to PHP variables for simplicity. We might confuse folks otherwise when we don't have $ref support. Let me know if y'all had more feedback on this. I'm happy to iterate.

Reply:
Yes, `$ref` would be ideal in the long run, but using PHP variables should also help with the maintenance 👍🏻 I would leave the final call to @TimothyBJacobs though :)

Reply:
Let's stick with variables for now.